liblangutil/ErrorReporter.cpp

```diff
@@ -72,7 +110,7 @@ void ErrorReporter::error(Error::Type _type, SourceLocation const& _location, st
 	m_errorList.push_back(err);
 }
 
-void ErrorReporter::error(Error::Type _type, SourceLocation const& _location, SecondarySourceLocation const& _secondaryLocation, string const& _description)
+void ErrorReporter::error([[maybe_unused]] ErrorCase _error, Error::Type _type, SourceLocation const& _location, SecondarySourceLocation const& _secondaryLocation, string const& _description)
```

Comment:
The problem with `maybe_unused` is that if the parameter gets used in the end, the tag can still stay there, can it? I very much prefer not naming the parameter instead.